File: member-service/src/main/java/com/hedvig/memberservice/events/NameUpdatedEvent.java
package com.hedvig.memberservice.events;

import lombok.Value;

import java.util.HashMap;
import java.util.Map;

@Value
public class NameUpdatedEvent implements Traceable {
  public final Long memberId;
  public final String firstName;
  public final String lastName;

  public NameUpdatedEvent(Long memberId, String firstName, String lastName) {

    this.memberId = memberId;
    this.firstName = firstName;
    this.lastName = lastName;
  }

  @Override
  public Map<String, Object> getValues() {
    Map result = new HashMap();
    result.put("First name", firstName);
    result.put("Last name", lastName);
    return result;
  }
}

File: member-service/src/main/java/com/hedvig/memberservice/events/NameUpdatedEvent.kt
package com.hedvig.memberservice.events

class NameUpdatedEvent(
    override val memberId: Long,
    val firstName: String,
    val lastName: String
) : Traceable {
    override fun getValues() = mapOf(
        "First name" to firstName,
        "Last name" to lastName
    )
}
